spec/system/main/login_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+describe 'logging in', type: :system do
+  let(:simulate_login) do
+    # Go to root path
+    visit root_path
+    # Select option to log in using MarkUs
+    click_link(I18n.t('main.login_with', name: Settings.local_auth_login_name ||
+      I18n.t('main.local_authentication_default_name')), href: nil)
+    # Enter username and password
+    fill_in('user_login', with: user_name, id: 'user_login')
+    fill_in('user_password', with: 'x', id: 'user_password')
+    # Login to MarkUs
+    click_button(I18n.t('main.log_in'), name: 'commit')
+  end
+
+  context 'Instructor' do
```

Comment:
So in terms of the actual tests, there isn't a need here to distinguish between instructor/ta/student. For the purposes of login and checking redirection, they're all the same `StandardUser`, so you should just create one test for that user type.